src/panels/config/automation/dialog-new-automation.ts

```diff
@@ -42,11 +44,15 @@ class DialogNewAutomation extends LitElement implements HassDialog {
 
   @state() private _opened = false;
 
+  @state() private _mode: BlueprintDomain = "automation";
+
   @state() public blueprints?: Blueprints;
 
-  public showDialog(): void {
+  public showDialog(params: { mode: "script" | "automation" }): void {
```

Comment:
Use the same type as used in `show-dialog` (export it), so when the type is changed we only have to do it in one place.

Reply:
Sorry I sort of knew I should have done this... but I tried to take the lazy route 😅 
Hopefully got it right this time, updated.